File: php/src/FactoryMethod/PracticePaymentI/Factories/PaymentFactory.php
<?php

namespace Ibrahim13\PdPhp\FactoryMethod\PracticePaymentI\Factories;

use Ibrahim13\PdPhp\FactoryMethod\PracticePaymentI\PaymentMethod\IPaymentMethod;

abstract class PaymentFactory {

    abstract public function createPaymentMethod(): IPaymentMethod;

	public function proccessPayment(int $amount) :string {

		$paymentMethod = $this->createPaymentMethod();

		 return $paymentMethod->pay($amount); 
	}
}

File: python/factory_method/practice_payment/factories/payment_factory.py
from abc import ABC, abstractmethod
from factory_method.practice_payment.payment_method.payment_method_contract import PaymentMethodContract


class PaymentFactory(ABC):

    @abstractmethod
    def create_payment_method(self) -> PaymentMethodContract:
        pass

    def process_payment(self, amount) -> str:
        payment_process = self.create_payment_method()
        return payment_process.pay(amount)
